src/main/java/frc/robot/commands/DriveWithDistance.java

```diff
@@ -0,0 +1,196 @@
+
+  }
+
+  /**
+   * Assigns the current conversion from encoder ticks to distance type
+   *
+   * @param distanceType    The distance unit that the robot is using for driving
+   */
+  private void assignDistanceType(DistanceType distanceType) {
+    //Setting the current conversion type according to the chosen distance unit
+    switch (distanceType) {
+      case FEET:
+        this.currentConversion = DriveBase.DistanceUnit.FEET.encoderTicks;
+
+        break;
+      case METERS:
+        this.currentConversion = DriveBase.DistanceUnit.METERS.encoderTicks;
+
+        break;
+      case CENTIMETERS:
+        this.currentConversion = DriveBase.DistanceUnit.CENTIMETERS.encoderTicks;
+
+        break;
+      case INCHES:
+        this.currentConversion = DriveBase.DistanceUnit.INCHES.encoderTicks;
+
+        break;
+
+      //Default is meters
+      default:
+
+        this.currentConversion = DriveBase.DistanceUnit.METERS.encoderTicks;
+        break;
+
+    }
+  }
+
+  /**
+   * Called  one when the command is initially scheduled and determines the desired distance to travel
+   * 
+   */
+  @Override
+  public void initialize() {
+
+    //Add the desired amount of encoder ticks to the current distance to get the amount of encoder ticks that the robot needs to drive
+    desiredRightDistance = driveBase.getRightDistance() + desiredRightEncoderTicks;
+    desiredLeftDistance = driveBase.getLeftDistance() + desiredLeftEncoderTicks;
+
+    addRequirements(DriveBaseHolder.getInstance());
+  }
+
+  /**
+   * Called every time the scheduler runs while the command is scheduled. (run every 20ms) and determines the current distance
+   * 
+   */
+  @Override
+  public void execute() {
+
+    //Get the current distance of the right encoder and store value in variable
+    double currentRightDistance = DriveBaseHolder.getInstance().getRightDistance();  
+    double currentLeftDistance = DriveBaseHolder.getInstance().getLeftDistance();
+
+    //Run motors according to output of the speeds
+    driveBase.tankDrive(leftSpeed, rightSpeed, false);
+
+    //Dashboard
+    SmartDashboard.putNumber("Left Speed", leftSpeed);
+    SmartDashboard.putNumber("Right Speed", rightSpeed);
+    SmartDashboard.putNumber("Left Error", desiredLeftDistance - currentLeftDistance);
+    SmartDashboard.putNumber("Right Error", desiredRightDistance - currentRightDistance);
+
+    //Code for figuring out when the robot is finished driving and handeling user input validation
+
+    // Update to the current right/left distance that the robot has driven
+    currentRightDistance = DriveBaseHolder.getInstance().getRightDistance();
+    currentLeftDistance = DriveBaseHolder.getInstance().getLeftDistance();
+
+    // If the robot has reached or surpassed the desired distance, then stop the
+    // robot. Otherwise, keep moving (moving forward OR moving backwards).
+    if (Math.abs(currentRightDistance) >= desiredRightDistance) {
```

Comment:
Question: as soon as either left or right are 'far enough' wouldn't we want to stop both anyway to avoid turning?

Reply:
No, we would want it to turn in that scenario. Depending on the constructor the user uses, the robot may have a different "desired distance" for the left and right motor.